### PR TITLE
FEAT: Support Django 5.1 - Remove index_together usage and update CI/Test Configs

### PR DESCRIPTION
**Pull request by NandanaRaol**
This pull request introduces support for Django 5.1, updates compatibility checks for removed `index_together`, and adjusts testing and dependency configurations accordingly. 
**Document link:**  [TASK1 Removing Dependency On Index_together().loop](https://microsoft.sharepoint.com/:fl:/g/contentstorage/CSP_b7c217ea-274e-4dad-9f29-6a25ac50eb56/EYUlB9bRbz9BkUFwGgSdP3wBa4KYJ7BS6al9sCUMg6szFg?e=aE42wN&nav=cz0lMkZjb250ZW50c3RvcmFnZSUyRkNTUF9iN2MyMTdlYS0yNzRlLTRkYWQtOWYyOS02YTI1YWM1MGViNTYmZD1iJTIxNmhmQ3QwNG5yVTJmS1dvbHJGRHJWZzF3cHo0ZUhHRkZwQk51eFg0WlVTZHBVQ2lEYVlTTFQ2aURLdXJSWmdpcSZmPTAxSFEyUk9NTUZFVUQ1TlVMUEg1QVpDUUxRRElDSjJQMzQmYz0lMkYmYT1Mb29wQXBwJnA9JTQwZmx1aWR4JTJGbG9vcC1wYWdlLWNvbnRhaW5lciZ4PSU3QiUyMnclMjIlM0ElMjJUMFJUVUh4dGFXTnliM052Wm5RdWMyaGhjbVZ3YjJsdWRDNWpiMjE4WWlFMmFHWkRkREEwYm5KVk1tWkxWMjlzY2taRWNsWm5NWGR3ZWpSbFNFZEdSbkJDVG5WNFdEUmFWVk5rY0ZWRGFVUmhXVk5NVkRacFJFdDFjbEphWjJseGZEQXhTRkV5VWs5TlRFRkpVMUZKV0VKQk5saFdRMXBLVjFsQ1JVcFVWRXRRVjFvJTNEJTIyJTJDJTIyaSUyMiUzQSUyMmM5NmMyOTM2LWQ5MzAtNGZjMC05ODMxLTk1NDNmMDViMDY3MSUyMiU3RA%3D%3D)
**Ado link:** https://sqlclientdrivers.visualstudio.com/mssql-django/_workitems/edit/37632

### CODE REVIEW

File: mssql/schema.py [hunk 1]
```diff
@@ -292,17 +292,25 @@ def alter_unique_together(self, model, old_unique_together, new_unique_together)
...
         Return a list of all index SQL statements (field indexes,
-        index_together, Meta.indexes) for the specified model.
+         Meta.indexes) for the specified model.
         """
         if not model._meta.managed or model._meta.proxy or model._meta.swapped:
             return []
         output = []
         for field in model._meta.local_fields:
             output.extend(self._field_indexes_sql(model, field))
+        # meta.index_together is removed in Django 5.1, so add a version check to handle compatibility
```
[the remark this answers is not in the document]
Comment (bewithgaurav) [reply mssql/schema.py]: not important, will resolve this

**Review by bewithgaurav — APPROVED** [review]
lgtm, can we rename this PR to something like this - since there's no "replacement"?
`FEAT: Support Django 5.1 - Remove index_together usage and update CI/Test Configs`

File: mssql/schema.py [hunk 2]
```diff
@@ -292,17 +292,24 @@ def alter_unique_together(self, model, old_unique_together, new_unique_together)
...
-            output.append(self._create_index_sql(model, fields, suffix="_idx"))
+         # meta.index_together is removed in Django 5.1, so add a version check to handle compatibility
+        if django_version < (5, 1):
+            # Iterate over each set of field names defined in index_together
+            for field_names in model._meta.index_together:
+               # Get the actual field objects for each field name
+               fields = [model._meta.get_field(field) for field in field_names]
+               # Generate the SQL statement to create the index for these fields
+               sql=self._create_index_sql(model, fields, suffix="_idx")
+               # If SQL was generated (not None), add it to the output list
```
Comment (bewithgaurav) [inline mssql/schema.py:310]: This is a small fix which handles edge cases
- If the generated SQL is not None, then only add it to output list.

File: testapp/tests/test_indexes.py [hunk 3]
```diff
@@ -116,9 +116,10 @@ def test_correct_indexes_exist(self):
                 expected_index_causes = []
                 if field.db_index:
                     expected_index_causes.append('db_index=True')
-                for field_names in model_cls._meta.index_together:
-                    if field.name in field_names:
-                        expected_index_causes.append(f'index_together[{field_names}]')
+                if VERSION < (5, 1):
```
Comment (bewithgaurav) [inline testapp/tests/test_indexes.py:119]: Added one task to have testcase for Meta.Indexes usage for v5.1
